packages/block-editor/src/components/index.js

```diff
@@ -42,6 +42,7 @@ export {
 export { default as ColorPalette } from './color-palette';
 export { default as ColorPaletteControl } from './color-palette/control';
 export { default as ContrastChecker } from './contrast-checker';
+export { default as DateFormatPicker } from './date-format-picker';
```

Comment:
This seems like a very generic component, so I wonder if `block-editor` is the right package.

Most of the other block editor components are either connected to data or very block specific.

Reply:
Yeah I umm'd and ahh'd about `@wordpress/block-editor` versus `@wordpress/components`. What do you think?

Reply:
I'm leaning towards `components`. I think it's ok to ship it in block-editor in this PR, but maybe don't make it stable to start with, so that it's still ok to move it later.

As a components package maintainer, it'd be good to get some thoughts from @ciampo about this one.

Reply:
Thank you for the ping!

TL;DR: We could consider moving it to the `@wordpress/components` package, although I suggest doing that in a separate PR. I 100% with @talldan 's suggestion about marking the component _unstable_ for the time being.

---

A few more detailed considerations:

- it's true that this component is generic for the `block-editor` package, so it would make sense to move it somewhere else. In case we moved it to `@wordpress/components`, though, we'd need to tweak the APIs a bit to make even more generic and decouple the component a bit more from the block editor (e.g. `siteFormat` could probably become `defaultFormat` ?)
- in case we moved the component to `@wordpress/components`, we'd need to refactor it:
    - from JS to TypeScript
    - from sass to CSS-in-JS (currently emotion)
    - add unit tests
    - add Storybook examples
    - ...and more in general, follow these [guidelines](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CONTRIBUTING.md)

So probably the best thing is to introduce the component in the `block-editor` package as experimental, then move it to `@wordpress/components` and refactor it as describe above in a separate, follow-up PR.

Also cc'ing @mirka (who also helps maintaining the components package) for any additional thoughts.

Reply:
Thanks @ciampo!

> So probably the best thing is to introduce the component in the block-editor package as experimental, then move it to @wordpress/components and refactor it as describe above in a separate, follow-up PR.

I shall do exactly this.